[Win] Reimplement LocalizedStrings.cpp without CF

fujii opened the pull request:
#### 62c199983aead46a9bcef25c66a41071713ceddd
<pre>
[Win] Reimplement LocalizedStrings.cpp without CF
<a href="https://bugs.webkit.org/show_bug.cgi?id=252728">https://bugs.webkit.org/show_bug.cgi?id=252728</a>

Reviewed by Alex Christensen.

Implement formatLocalizedString with _vsnwprintf. Some functions are
called only by Cocoa ports. Conditioned out them with #if
PLATFORM(COCOA).

* Source/WebCore/platform/LocalizedStrings.cpp:
(WebCore::formatLocalizedString):
(WebCore::webCoreBundle):
(WebCore::copyLocalizedString):
(WebCore::localizedString):
(WebCore::imageTitle):
(WebCore::validationMessageRangeUnderflowText):
(WebCore::validationMessageRangeOverflowText):
* Source/WebCore/platform/LocalizedStrings.h:

Canonical link: <a href="https://commits.webkit.org/260705@main">https://commits.webkit.org/260705@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/da6f224c0cd51423a1827a135d2667e9598f8254

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/109106 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/18189 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/41928 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/641 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/118365 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/112991 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/19647 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/9490 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/101347 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/114865 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/14730 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/97966 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/42898 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/96702 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/29620 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/84631 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/10970 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/30963 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/11707 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/7893 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/17076 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/50566 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7387 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/13315 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

donny-dont commented on Source/WebCore/platform/LocalizedStrings.cpp:
```diff
@@ -107,11 +113,7 @@ RetainPtr<CFStringRef> copyLocalizedString(CFStringRef key)
-#if PLATFORM(WIN)
-    CFBundleRef bundle = webKitBundle();
-#else
     CFBundleRef bundle = webCoreBundle();
```
It doesn't look like you need the temporary here.

fujii replied:
Fixed.

donny-dont reviewed (commented):
I'd like someone from 🍎 to provide the final approval since I'm not too sure about the changes to the guards. Feels like you're mixing `USE(CF)` and `PLATFORM(COCOA)` and things should be more consistent here.

fujii committed:
[Win] Reimplement LocalizedStrings.cpp without CF
https://bugs.webkit.org/show_bug.cgi?id=252728

Reviewed by Alex Christensen.

Implement formatLocalizedString with _vsnwprintf. Some functions are
called only by Cocoa ports. Conditioned out them with #if
PLATFORM(COCOA).

* Source/WebCore/platform/LocalizedStrings.cpp:
(WebCore::formatLocalizedString):
(WebCore::webCoreBundle):
(WebCore::copyLocalizedString):
(WebCore::localizedString):
(WebCore::imageTitle):
(WebCore::validationMessageRangeUnderflowText):
(WebCore::validationMessageRangeOverflowText):
* Source/WebCore/platform/LocalizedStrings.h:

Canonical link: https://commits.webkit.org/260705@main
